TodoApp.ConsoleApp/Services/TodoService.cs

```diff
@@ -30,21 +30,30 @@ public Task GetByID(Guid id)
 
         public bool Update(Task task)
```

Comment:
Any argument validation should be put at the top of the method, and the error handling should be done there. This way the method depth is less (one less `if` in this case).
`ArgumentNullException` (and every other Argument exception) require argument name as first parameter, not the error message. Use the [nameof operator](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/nameof).

Not sure why you added the explicit argument when calling `Delete`. You can skip it.

For Example:
```csharp
public bool Update(Task task)
{
    if (task == null)
    {
        throw new ArgumentNullException(nameof(task));
    }

    bool isDeleted = Delete(task.ID);
            
    if (!isDeleted)
    {
        return false;
    }

    var newTask = Create(task);
    return newTask != null;
}
```